src/components/modals/Username.js

```diff
@@ -44,17 +49,28 @@ class UsernameModal extends Component {
 
   isUsernameValid(username) {
     if (!username || username.length === 0) {
+      this.setState({
+        displayAlert: true,
+        failMessage: 'Must Not Be Empty'
+      });
+      return false;
+    }
+    else if (username.length > 45) {
+      this.setState({
+        displayAlert: true,
+        failMessage: 'Length May Not Exceed 45'
```

Comment:
@elimisteve `username` variable is not stored in state, it's being passed around methods. 
Option: Refactor - store username (length) in state and have a function that returns text, like this: 
``` 
  failMessage = () => {
    if (this.state.displayAlert && this.state.username.length = 0) {
      return 'Must not be empty';
    }
    else if (this.state.displayAlert && this.state.username.length > 45) {
      return 'Length must not exceed 45';
    }
  }
```